tests/posix/common/src/pthread_attr.c

```diff
@@ -441,12 +441,25 @@ ZTEST(pthread_attr, test_pthread_attr_setstacksize)
 
 ZTEST(pthread_attr, test_pthread_attr_large_stacksize)
 {
+	int ret;
 	size_t actual_size;
 	const size_t expect_size = BIT(CONFIG_POSIX_PTHREAD_ATTR_STACKSIZE_BITS);
 
-	zassert_ok(pthread_attr_setstacksize(&attr, expect_size));
-	zassert_ok(pthread_attr_getstacksize(&attr, &actual_size));
-	zassert_equal(actual_size, expect_size);
+	ret = pthread_attr_setstacksize(&attr, expect_size);
+	switch (ret) {
+	case 0:
+		zassert_ok(pthread_attr_getstacksize(&attr, &actual_size));
+		zassert_equal(actual_size, expect_size);
+		break;
+	case ENOMEM:
+		TC_PRINT("Failed to allocated thread stack of size %zu\n", expect_size);
```

Comment:
Just a note, this attempts to allocate an 8MB thread stack (which would predictably fail in most cases).

I don't know if there is necessarily a more graceful way to do this.